program/src/onchain.rs

```diff
@@ -26,36 +26,50 @@ pub fn invoke_transfer_checked<'a>(
     additional_accounts: &[AccountInfo<'a>],
     amount: u64,
     decimals: u8,
+    fee: Option<u64>,
     seeds: &[&[&[u8]]],
-) -> ProgramResult {
-    let mut cpi_instruction = instruction::transfer_checked(
-        token_program_id,
-        source_info.key,
-        mint_info.key,
-        destination_info.key,
-        authority_info.key,
-        &[], // add them later, to avoid unnecessary clones
```

Comment:
The source of the bug. Signer status is added depending if the accounts are added here. See: https://github.com/solana-program/token-2022/blob/d52d1cd18228888ce31f7e9ba47f436b1117b82c/program/src/instruction.rs#L1689-L1692